.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,41 @@
+name: Publish Docker image
+on:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - '*'
+    paths-ignore:
+      - '.gitignore'
+      - '**.md'
+      - 'test/'
+
+jobs:
+  build_and_publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+      - name: Extract branch name
+        if: startsWith(github.ref, 'refs/tags/')
+        run: echo "##[set-output name=branch;]${GITHUB_REF#refs/heads/}"
```

Comment:
oops, this is getting the wrong thing, gotta look at `refs/tags` and set the tag, not the branch name